src/contracts/core/RewardsCoordinator.sol

```diff
@@ -284,16 +294,73 @@ contract RewardsCoordinator is
     }
 
     /// @inheritdoc IRewardsCoordinator
-    function setOperatorAVSCommission(address operator, address avs, uint16 commission) external {
+    function setOperatorAVSCommission(
+        address operator,
+        address avs,
+        uint16 commission
+    ) external onlyWhenNotPaused(PAUSED_OPERATOR_AVS_COMMISSION) {
         require(msg.sender == operator, "RewardsCoordinator.setOperatorAVSCommission: caller is not the operator");
-        //TODO: Add a check to ensure that the operator is registered to the AVS.
-        //TODO: Add a 7 day delay.
+        require(
+            commission <= ONE_HUNDRED_IN_BIPS,
+            "RewardsCoordinator.setOperatorAVSCommission: commission must be <= 10000 bips"
+        );
+        uint32 activatedAt = SafeCast.toUint32(block.timestamp) + activationDelay;
+
+        OperatorCommission storage operatorCommission = operatorAVSCommissionBips[operator][avs];
+
+        // If, the earlier 'new' commission is activated, we update the 'old' commission with the earlier 'new' commission.
+        // Else, the earlier 'old' commission remains the same. This is essentially resetting the activation delay window
+        // since the earlier commission setting didn't complete.
+        if (block.timestamp >= operatorCommission.activatedAt) {
+            operatorCommission.oldCommissionBips = operatorCommission.newCommissionBips;
+        }
+        operatorCommission.newCommissionBips = commission;
+        operatorCommission.activatedAt = activatedAt;
+
+        emit OperatorAVSCommissionBipsSet(
+            msg.sender,
+            operator,
+            avs,
+            activatedAt,
+            operatorCommission.oldCommissionBips,
+            commission
+        );
+    }
+
+    /// @inheritdoc IRewardsCoordinator
+    function setOperatorPICommission(
+        address operator,
+        uint16 commission
+    ) external onlyWhenNotPaused(PAUSED_OPERATOR_PI_COMMISSION) {
+        require(msg.sender == operator, "RewardsCoordinator.setOperatorPICommission: caller is not the operator");
+        require(
+            commission <= ONE_HUNDRED_IN_BIPS,
+            "RewardsCoordinator.setOperatorPICommission: commission must be <= 10000 bips"
+        );
+        require(
+            commission >= MIN_PI_COMMISSION_BIPS,
+            "RewardsCoordinator.setOperatorPICommission: commission must be >= 1000 bips"
+        );
+        uint32 activatedAt = SafeCast.toUint32(block.timestamp) + activationDelay;
 
-        OperatorAVSCommission storage operatorAVSCommission = operatorAVSCommissionBips[operator][avs];
+        OperatorCommission storage operatorCommission = operatorPICommissionBips[operator];
```

Comment:
whats the reasoning for a different map as opposed to a magic PI AVS?

Reply:
Also PI has the specific requirement of having a minimum 10% commission.